src/shortcuts/__tests__/undo.ts

```diff
@@ -0,0 +1,126 @@
+import { importText } from '../../action-creators'
+import { RANKED_ROOT, ROOT_TOKEN } from '../../constants'
+import { exportContext } from '../../selectors'
+import { createTestStore } from '../../test-helpers/createTestStore'
+
+it('Undo thought change', async () => {
+
+  const store = createTestStore()
+
+  // import thoughts
+  store.dispatch(importText(RANKED_ROOT, `
+  - a
+  - b`))
+
+  store.dispatch([{
+    type: 'setCursor', thoughtsRanked: [
+      { value: 'a', rank: '0' }
+    ]
+  }, {
+    type: 'existingThoughtChange',
+    newValue: 'aa',
+    oldValue: 'a',
+    context: [ROOT_TOKEN],
+    thoughtsRanked: [{ value: 'a', rank: 0 }]
+  }, {
+    type: 'undoAction'
+  }])
+
+  const exported = exportContext(store.getState(), [ROOT_TOKEN], 'text/plain')
+
+  const expectedOutput = `- ${ROOT_TOKEN}
+  - a
+  - b`
+
+  expect(exported).toEqual(expectedOutput)
+})
+
+it('Group all navigation actions preceding a thought change and undo them together', async () => {
+
+  const store = createTestStore()
+
+  store.dispatch([importText(RANKED_ROOT, `
+  - a
+  - b
+  - c
+  - d`), {
+    type: 'cursorDown'
+  }, {
+    type: 'existingThoughtChange',
+    newValue: 'aa',
+    oldValue: 'a',
+    context: [ROOT_TOKEN],
+    thoughtsRanked: [{ value: 'a', rank: 0 }]
+  }, {
+    type: 'cursorDown'
+  }, {
+    type: 'cursorDown'
+  }, {
+    type: 'setCursor'
+  }, {

```

Comment:
Use `thoughtsRanked: null` explicitly with `setCursor`. `undefined` is not a valid value for `cursor`.